test/Schema.js

```diff
@@ -1190,4 +1190,46 @@ describe('Schema tests', function (){
     });
     done();
   });
+
+  it('Should throw error when type is map but no map is provided', function (done) {
+    try {
+      var schema = new Schema({
+        race: {
+          type: 'map',
+        }
+      });
+    } catch (err) {
+      err.should.be.instanceof(Error);
+      err.should.be.instanceof(errors.SchemaError);
```

Comment:
Not sure I'm a huge fan of this. If there is no error thrown, this test will still pass. I think it'd be better to create a variable above (set to nothing by default). Then inside the catch, set `err` to that variable we created above. Then right before `done` is called, run the assertions, to ensure that variable we created exists, ensure it's an instance of, etc.